zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZkClientUriDomainMappingHelper.java

```diff
@@ -79,8 +79,15 @@ public ZkClientUriDomainMappingHelper(ZooKeeperServer zks) {
     }
 
     if (zks.getZKDatabase().getNode(rootPath) == null) {
-      throw new IllegalStateException(
-          "ZkClientUriDomainMappingHelper::ClientUriDomainMapping root path does not exist :" + rootPath);
+      try {
+        zks.getZKDatabase().getDataTree().createNode(rootPath, new byte[0],
+            ZooDefs.Ids.READ_ACL_UNSAFE, -1L, -1,
```

Comment:
Question: is this the appropriate ACL the root path should have?

Reply:
I was thinking we don't want to set the ACL to a specific domain at this moment (in code) yet, since the domain name can change. So the behavior I'm expecting is that: the root path is open read at the start-up time, then superUserId can manually modify the ACL later to the admin domain.

Reply:
Hmm.. in that case did you mean to use `OPEN_ACL_UNSAFE`, not `READ_ACL_UNSAFE` since you want to allow writes?

Also, is there a way to make sure that the user follows the expected operations?

Reply:
`OPEN_ACL_UNSAFE` is not the right one I think, because any user can create/modify nodes before the ACL is changed to allow zk admin only. So I used `READ_ACL_UNSAFE` (super user can still write even though the node has READ ACL) to make sure no one can change this node before zk admin thinks it's ready and change the ACL to allow write.

Reply:
@narendly Please let me know if it makes sense to you. I added some comments there so users would know that they need to manually set ACLs to this path later.

Reply:
I think we could make it work either way. But we should consider user experience. What would be easier for users/ZK server admins? Do we have a clear case from the UX perspective?

Reply:
neat. can we also add some brief comment about the arguments for createNode? It would be easier to understandat is "-1L, -1, 0L and 0L"

Reply:
@narendly I think you have a point here so I changed it to `OPEN_ACL_UNSAFE`, this will provide better ux.
@rahulrane50 I added some comments for createNode